andhow-core/src/test/java/org/yarnandtail/andhow/internal/ConstructionProblemTest.java

```diff
@@ -0,0 +1,165 @@
+package org.yarnandtail.andhow.internal;
+
+import static org.junit.Assert.*;
+import static org.mockito.Mockito.*;
+import org.junit.Test;
+import org.yarnandtail.andhow.api.*;
+import org.yarnandtail.andhow.AndHowInit;
+import java.util.ArrayList;
+import java.util.List;
+import org.yarnandtail.andhow.AndHow;
+
+
+
+public class ConstructionProblemTest {
+
+    @Test    
+    public void testNoUniqueNames() {
+        GroupProxy refGroup = mock(GroupProxy.class);
+        
+        Property refProperty = mock(Property.class);
+        GroupProxy badGroup = mock(GroupProxy.class);
+        Property badProperty = mock(Property.class);
+        String conflictName = "conflict name test";
+        ConstructionProblem.NonUniqueNames instance = new ConstructionProblem.
+        NonUniqueNames(refGroup, refProperty, badGroup, badProperty, conflictName);
+        
```

Comment:
When there is an issue with how a Property is constructed, there should be a `GroupProxy badGroup` and a `Property badProperty` in the `ConstructionProblem` constructor that points the user to exactly where the problem is.   **_Group_** in this case would refer to some class in the user's application, **_Property_** refers to an AndHow Property in that class.  The error messages end up looking something like this:
`The AndHow Property org.someapp.MyClass.SomeAndHowProperty has this problem...`

`ConstructionProblem` has a `getBadPropertyCoord()` that points to what Property has the problem when the problem is Property related:
```suggestion
        assertEquals(badGroup , instance.getBadPropertyCoord().getGroup());
        assertEquals(badProperty , instance.getBadPropertyCoord().getProperty());
```